src/internet_identity/src/main.rs

```diff
@@ -800,20 +841,28 @@ fn stats() -> InternetIdentityStats {
 // used both in init and post_upgrade
 fn init_assets() {
     use assets::ContentEncoding;
+    use assets::ContentType;
 
     STATE.with(|s| {
         let mut asset_hashes = s.asset_hashes.borrow_mut();
 
         ASSETS.with(|a| {
             let mut assets = a.borrow_mut();
-            assets::for_each_asset(|name, encoding, contents, hash| {
+            assets::for_each_asset(|name, encoding, content_type, contents, hash| {
                 asset_hashes.insert(name, *hash);
-                let headers = match encoding {
+                let mut headers = match encoding {
                     ContentEncoding::Identity => vec![],
                     ContentEncoding::GZip => {
                         vec![("Content-Encoding".to_string(), "gzip".to_string())]
                     }
                 };
+                let content_type = match content_type {
+                    ContentType::HTML => "text/html".to_string(),
+                    ContentType::JS => "text/javascript".to_string(),
+                    ContentType::ICO => "image/vnd.microsoft.icon".to_string(),
+                    ContentType::WEBP => "image/webp".to_string(),
+                };
```

Comment:
Would it be more elegant to implement this mapping directly on `ContentType`? E.g. like this: 
```
impl ContentType {
    fn to_mime_type_string(&self) -> String {
        match self {
            ContentType::HTML => "text/html".to_string(),
            ContentType::JS => "text/javascript".to_string(),
            ContentType::ICO => "image/vnd.microsoft.icon".to_string(),
            ContentType::WEBP => "image/webp".to_string(),
        }
    }
}
```

Reply:
Done